File: MPI/solutions/mpi_hello_world.f
      program hello

        use mpi
        implicit none

        integer :: id = 0
        integer :: size = 0
        integer :: ierr

        call MPI_INIT(ierr)
        call MPI_COMM_RANK(MPI_COMM_WORLD, id, ierr)
        call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierr)

        print *,"Hello world from process",id," of ",size

        call MPI_FINALIZE(ierr)

      end program

File: MPI/solutions/mpi_hello_world.c
#include <mpi.h>
#include <stdio.h>

void main(int argc, char* argv[])
{
  int id = 0;
  int size = 0;
  MPI_Init(&argc, &argv);
  MPI_Comm_rank(MPI_COMM_WORLD, &id);
  MPI_Comm_size(MPI_COMM_WORLD, &size);
  printf("Hello world from process %d of %d\n", id, size);
  MPI_Finalize();
}
